### PR TITLE
flawfinder: init at 2.0.15

### LINKED ISSUE #117382: flawfinder
**Project description**
Flawfinder is a simple program that scans C/C++ source code and reports potential security flaws. It can be a useful tool for examining software for vulnerabilities, and it can also serve as a simple introduction to static source code analysis tools more generally. It is designed to be easy to install and use. Flawfinder supports the Common Weakness Enumeration (CWE) and is officially CWE-Compatible.

**Metadata**

* homepage URL: http://dwheeler.com/flawfinder/
* source URL: https://github.com/david-a-wheeler/flawfinder
* license: gpl2
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Tool to examines C/C++ source code for security flaws

http://dwheeler.com/flawfinder

Related to #81418

Fixes #117382

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 117417` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>flawfinder</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117417` at 90894a7d run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>flawfinder</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/flawfinder/default.nix:23:3:
```
   |
23 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `coreutils`.
Near pkgs/development/tools/flawfinder/default.nix:3:3:
```
  |
3 | , coreutils
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `makeWrapper`.
Near pkgs/development/tools/flawfinder/default.nix:6:3:
```
  |
6 | , makeWrapper
  |   ^
```
</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 117417` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>flawfinder</li>
  </ul>
</details>

File: pkgs/development/tools/flawfinder/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+, installShellFiles
+, python3
+}:
+
+stdenv.mkDerivation rec {
+  pname = "flawfinder";
+  version = "2.0.15";
+
+  src = fetchurl {
+    url = "http://dwheeler.com/flawfinder/flawfinder-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/flawfinder/default.nix:13]:
```suggestion
    url = "https://dwheeler.com/flawfinder/flawfinder-${version}.tar.gz";
```

File: pkgs/development/tools/flawfinder/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    runHook preInstall
+    mkdir -p $out/bin
+    cp ${pname} $out/bin
+    installManPage flawfinder.1
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Tool to examines C/C++ source code for security flaws";
+    homepage = "http://dwheeler.com/flawfinder/";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/flawfinder/default.nix:31]:
```suggestion
    homepage = "https://dwheeler.com/flawfinder/";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/flawfinder/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/tools/flawfinder/default.nix]: Updated, thanks.

File: pkgs/development/tools/flawfinder/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/tools/flawfinder/default.nix]: Updated, thanks.

**Pull request merged**